GoogleForCreators/web-stories-wp
# Redesign: Snackbar

## Context
- Implement the Snackbar from the design system in the editor 


## Summary
- Delete old snackbar component, replace with design system
- change some styles and config to support the editor's use cases -> shorter min duration, allow dismissing on choosing the action, include `place`


## Relevant Technical Choices
- The snackbar already removed its own messages, so I removed the timeouts from the editor's provider
-  if you search`showSnackbar` in the `edit-story` code, that will show you all the interactions which cause snackbars. 


## To-do
- The figma shows a `Retry` button. We should include changes to update the components which use the snackbar to log error messages and include the retry functionality. I can probably do it in this PR but I wanted to keep it small to start.
- Some of the snackbars don't ever show (and didn't show up on `main` either), so I'm going to include a TODO to write bug tickets to fix those

- [x] write tickets to fix snackbar messages so they show when intended
- [x] also track tech debt to share a provider from the design system and change prop names



## User-facing changes
![Screen Shot 2021-03-09 at 3 44 38 PM](https://user-images.githubusercontent.com/41136059/110548946-95479400-80ee-11eb-8050-14effb42f83d.png)
![Screen Shot 2021-03-09 at 3 43 21 PM](https://user-images.githubusercontent.com/41136059/110548949-9678c100-80ee-11eb-8b05-db92cf2bc0d6.png)



## Testing Instructions



If you're pulling down the code, you can easily try out different configurations by changing the `showSnackbar` call in the `devTools` component and triggering the `copyToClipboard`. 

### QA






This PR can be tested by following these steps:

- In order to see snackbars QA will have to simulate some errors. Some of these can be done by network throttling and some of them can be triggered. The snackbar notifications appear in the editor at these times:

1. Open dev tools (cmd+shift+opt+j) and click "Copy". A snackbar should appear briefly. It should not have a dismiss option or an action button. _(note: load Story Snackbar doesn't show on main, will write a ticket to address)_
2. Drag-and-drop a `.mov` file into upload media. Multiple snackbars with close buttons may appear-- one that shows Video optimization is in progress and one that indicates a failure. When I tried an `mov` file it failed every time and saw both snackbars. If the transcoding works there will be no failure snackbar.

The following can be shown by simulating network errors.
3. Snackbar with close button appears when media upload fails.
5. Snackbar with close button appears when uploaded media deletion fails.
6. Snackbar with close button appears when fetching 3p media fails
7. Snackbar with close button appears when fetching 3P media categories fails.
8. Snackbar with close button appears when saving story fails (click "Save" in the upper-right hand corner of the editor).

All snackbars in the Dashboard should have a close button and have the dark theme.
All snackbars in the Editor (except the dev tools copy to clipboard) should have a close button and the light theme.
No snackbars should have a `Retry` button as of this change.
All Snackbars should appear at the bottom & in the middle of the page.
In the Dashboard, one snackbar should show at a time. More snackbars will fade away.

### UAT








## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
No


### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #6521
